packages/core/src/components/steps-modal/steps-modal.tsx

```diff
@@ -85,6 +95,13 @@ export class AtomStepsModal {
   }
 
   private handleSecondaryClick = () => {
+    if (this.currentStep === this.customInitialStep) {
+      this.isOpen = false
+      this.atomCancel.emit()
+
```

Comment:
Guess it is not a steps-modal responsibility :thinking:

Reply:
yeah, i forgot that  when is on first step and clicks secondary It should only emit cancel